cli/command/cli.go

```diff
@@ -166,6 +167,16 @@ func (cli *DockerCli) RegistryClient(allowInsecure bool) registryclient.Registry
 // Initialize the dockerCli runs initialization that must happen after command
 // line flags are parsed.
 func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions) error {
+	cliflags.SetLogLevel(opts.Common.LogLevel)
```

Comment:
I'm wondering why these changes are needed 🤔 ? Are they relevant for this PR? Or maybe do they fix a bug? In that case move it to another PR?

Reply:
See the first commit's commit message (7a9817d5b6476e1ee49c81f7861edc5605ed85fe):
```
    Fold `dockerPreRun` into `DockerCli.Initialize`
    
    All of the current callers follow the pattern:
    
        dockerPreRun(opts)
        err := dockerCli.Initialize(opts) ...
    
    So there is no semantic change into merging the content of `dockerPreRun` into the head of `Initialize`.
    
    I'm about to add a new caller outside of the `cmd/docker` package and this
    seems preferable exporting `DockerPreRun`.
    
    Signed-off-by: Ian Campbell <ijc@docker.com>
```

Reply:
Right, seems legit 😄cc @thaJeztah / @vdemeester

Reply:
@thaJeztah @vdemeester any opinions?

Reply:
Seems sensible; especially:

> this seems preferable exporting `DockerPreRun`.


(Even wondering if we need a separate `Initialize`, and if it should be part of the constructor)

Reply:
It needs be called separately from `PersistentPreRunE` because it needs to consume the parsed options but the `dockerCli` instance needs to exist before that so it can be captured into other scopes, such as `RunE`.

Reply:
@vdemeester just spotted this https://godoc.org/github.com/spf13/cobra#OnInitialize
You should give it a try ?

Reply:
Some of the callers of this `Initialize` might be able to use `OnInitiliaze` (although it doesn't allow for errors to be returned) but I think (from code inspection only) there are other paths which would still need to manually call `Initialize`, at least `FlagErrorFunc` (I'm fairly confident) and possibly (I'm not sure) the `HelpFunc` stuff so I think it doesn't buy us a lot.